pkg/kv/kvserver/replica.go

```diff
@@ -1489,8 +1494,7 @@ func (r *Replica) checkSpanInRangeRLocked(ctx context.Context, rspan roachpb.RSp
 		return nil
 	}
 	return roachpb.NewRangeKeyMismatchError(
-		ctx, rspan.Key.AsRawKey(), rspan.EndKey.AsRawKey(), desc, r.mu.state.Lease,
```

Comment:
Drive-by comment — I intend to do a deeper review later: Isn't the replica already `RLocked` at this point? So won't this deadlock? Go's locks are not re-entrant, though this only result in a deadlock for re-entrant attempts on RLock if there was an intermediate call to Lock.

Reply:
That seems right - thanks for pointing this out. It is trying to re-acquire a read lock in `GetClosedTimestampPolicy`. Updating.